projects/components/src/filter-bar/filter-bar.component.ts

```diff
@@ -129,12 +118,20 @@ export class FilterBarComponent implements OnChanges {
 
     this.filtersChange.emit(this.internalFilters);
 
+    if (this.syncWithUrl && this.hasAttributes()) {
+      this.writeToUrlFilter();
+    }
+  }
+
+  private readFromUrlFilters(): void {
     if (this.syncWithUrl) {
-      this.publishToUrlFilter();
+      this.onFiltersChanged(this.filterBarService.getUrlFilters(this.attributes || []));
```

Comment:
Read should only be called if syncWithUrl is true (which is what we do for write, so they behave differently)